reference/matrix/csr_kernels.cpp

```diff
@@ -165,6 +166,46 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_MOVE_TO_DENSE_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void convert_to_ell(std::shared_ptr<const ReferenceExecutor> exec,
+                    matrix::Ell<ValueType, IndexType> *result,
+                    const matrix::Csr<ValueType, IndexType> *source)
+{
+    const auto num_rows = source->get_size()[0];
+    const auto num_cols = source->get_size()[1];
+    const auto vals = source->get_const_values();
+    const auto col_idxs = source->get_const_col_idxs();
+    const auto row_ptrs = source->get_const_row_ptrs();
+
+    const auto num_stored_elements_per_row =
+        result->get_num_stored_elements_per_row();
+
+    for (size_type row = 0; row < num_rows; row++) {
+        for (size_type i = 0; i < num_stored_elements_per_row; i++) {
+            result->val_at(row, i) = zero<ValueType>();
+            result->col_at(row, i) = 0;
+        }
+        for (size_type col_idx = 0; col_idx < row_ptrs[row + 1] - row_ptrs[row];
+             col_idx++) {
+            result->val_at(row, col_idx) = vals[row_ptrs[row] + col_idx];
+            result->col_at(row, col_idx) = col_idxs[row_ptrs[row] + col_idx];
+        }
+    }
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_CSR_CONVERT_TO_ELL_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void move_to_ell(std::shared_ptr<const ReferenceExecutor> exec,
+                 matrix::Ell<ValueType, IndexType> *result,
+                 matrix::Csr<ValueType, IndexType> *source) GKO_NOT_IMPLEMENTED;
```

Comment:
We should definitively not have any `NOT_IMPLEMENTED` kernels inside reference, so I would strongly suggest to remove this kernel all together.